aptos-move/block-executor/src/view.rs

```diff
@@ -1495,19 +1601,25 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>> TResourceGroupView for Lat
     ) -> PartialVMResult<ResourceGroupSize> {
         let mut group_read = match &self.latest_view {
             ViewState::Sync(state) => state.read_group_size(group_key, self.txn_idx)?,
-            ViewState::Unsync(state) => state.unsync_map.get_group_size(group_key),
+            ViewState::Unsync(state) => match state.unsync_map.get_group_size(group_key) {
+                Some(group_size) => GroupReadResult::GroupSize(group_size),
+                None => GroupReadResult::Uninitialized,
+            },
         };
 
         if matches!(group_read, GroupReadResult::Uninitialized) {
             self.initialize_mvhashmap_base_group_contents(group_key)?;
 
             group_read = match &self.latest_view {
                 ViewState::Sync(state) => state.read_group_size(group_key, self.txn_idx)?,
```

Comment:
Maybe `read_group_size` on sync map can provide a similar API, it will also be easier to check that `into_group_size` invariant on having size variant holds.

Reply:
Yes GroupSize here is absolute overkill, and I think the ReadResult and GroupReadResult are generally pretty bad abstractions. But current refactors seem to be on the right path and show some of these improvement directions.